nemo/collections/asr/parts/preprocessing/features.py

```diff
@@ -401,32 +469,232 @@ def forward(self, x, seq_len):
         # log features if required
         if self.log:
             if self.log_zero_guard_type == "add":
-                x = torch.log(x + self.log_zero_guard_value_fn(x))
+                x = torch.log(x + self.log_zero_guard_value_fn(x.dtype))
             elif self.log_zero_guard_type == "clamp":
-                x = torch.log(torch.clamp(x, min=self.log_zero_guard_value_fn(x)))
+                x = torch.log(torch.clamp(x, min=self.log_zero_guard_value_fn(x.dtype)))
             else:
                 raise ValueError("log_zero_guard_type was not understood")
 
+        # mask to zero any values beyond seq_len in batch
+        max_len = x.size(-1)
+        mask = torch.arange(max_len).to(x.device)
+        mask = mask.expand(x.size(0), max_len) >= seq_len.unsqueeze(1)
+        x = x.masked_fill(mask.unsqueeze(1).type(torch.bool).to(device=x.device), self.pad_value)
+        del mask
+
         # frame splicing if required
         if self.frame_splicing > 1:
-            x = splice_frames(x, self.frame_splicing)
+            x = splice_frames(x, stacking=self.frame_splicing)
```

Comment:
Passing stacking but not subsampling?

Reply:
I did not apply subsampling because the previous implementation didn't do it either. I think the particular stacking/subsampling settings might depend on the particular model. It can be easily extended to support arbitrary subsampling when there is a need.